packages/ember-htmlbars/lib/keywords/closure-component.js

```diff
@@ -117,6 +117,6 @@ function getPositionalParams(container, componentPath) {
   }
 }
 
-export function mergeHash(original, updates) {
-  return assign(original, updates);
+export function mergeInNewHash(original, updates) {
+  return assign(Object.create(original), updates);
```

Comment:
Do we need object inheritance here? Would the following also work:

    assign({}, original, updates);